modules/mesaides/templates/automation_script.erb

```diff
@@ -16,8 +16,16 @@ do
     mv --backup --suffix=$(date '+.bck_upto_%Y-%m-%d_%H-%M-%S') $tmp_path $head_path
 done
 
+function run_puppet {
+    set +e
+    eval "puppet apply --detailed-exitcodes --verbose $*"
+    exit_code=$?
+    set -e
+    [ $exit_code -eq 0  ] || [ $exit_code -eq 2 ]
```

Comment:
If this has to be made in several files, it could be worth making it a standalone wrapper script.